src/InputGroup/_InputGroupAddon.js

```diff
@@ -0,0 +1,74 @@
+import classnames from 'classnames';
+import Icon from '../Icon/Icon';
+import { INPUT_GROUP_ADDON_TYPES } from '../utils/constants';
+import PropTypes from 'prop-types';
+import React from 'react';
+import withStyles from '../utils/WithStyles/WithStyles';
+
+const InputGroupAddon = ({
+    addon,
+    actions,
+    addonClassNames,
+    children,
+    className,
+    compact,
+    disableStyles,
+    glyph,
+    ...otherProps
+}) => {
+
+    let returnObject = children;
+
+    if (glyph && addon === 'icon') {
+        returnObject = (
+            <Icon
+                disableStyles={disableStyles}
+                glyph={glyph}
+                role='presentation' />
+        );
+    }
+
+    children = React.Children.map(children, (child) => {
+        return React.cloneElement(child, {
+            actions: actions,
+            compact: compact
+        });
+    });
+
+    const addonClasses = classnames(
+        addonClassNames,
+        'fd-input-group__addon',
+        [{ 'fd-input-group__addon--button': !!actions || addon === 'button' }],
+        [{ 'fd-input-group__addon--button--compact': compact && addon === 'button' }],
+    );
+
+    return (
+        <span className={addonClasses} {...otherProps} >
+            {returnObject}
+        </span>
+    );
+};
+
+InputGroupAddon.displayName = 'InputGroupAddon';
+
+InputGroupAddon.propTypes = {
+    actions: PropTypes.bool,
+    addon: PropTypes.oneOf(INPUT_GROUP_ADDON_TYPES),
```

Comment:
Might be able to just get away with a boolean `isButton` prop.  There's really nothing special we would do with any other children.